src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -223,8 +272,8 @@ const PreviewHeader = ({ queryParams, cluster, readOnlyAccess, refreshClusters,
         !mode || clusterStatus === null || clusterStatus === 'Stopped', () => h(Fragment, [
           h(ButtonSecondary, {
             style: { paddingRight: '1rem', paddingLeft: '1rem', paddingTop: '1rem', paddingBottom: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
-            onClick: () => locked ? setFileInUseOpen(true) : chooseMode('edit')
-          }, [icon(locked ? 'lock': 'edit', { style: { paddingRight: '3px' } }), locked ? 'EDIT (IN USE)' : 'EDIT']),
+            onClick: () => welderEnabled ? (locked ? setFileInUseOpen(true) : chooseMode('edit')) : setEditModeDisabledOpen(true)
+          }, [icon(welderEnabled ? (locked ? 'lock': 'edit') : 'warning-standard', { style: { paddingRight: '3px' } }), welderEnabled ? (locked ? 'EDIT (IN USE)' : 'EDIT') : 'EDIT (DISABLED)']),
```

Comment:
It looks like there are three mostly-distinct states here, and the nested ternary operators are a little hard to follow. What about refactoring this into 3 separate buttons so there can be a single top-level `cond`?